metrics/prometheus_exporter_test.go

```diff
@@ -57,6 +61,71 @@ func TestNewPrometheusExporter(t *testing.T) {
 	}
 }
 
+func TestNewPrometheusExporter_fromEnv(t *testing.T) {
+	exporterOptions := ExporterOptions{
+		ConfigMap: map[string]string{},
+		Domain:    servingDomain,
+		Component: testComponent,
+	}
+	testCases := []struct {
+		name                   string
+		prometheusPortVarName  string
+		prometheusPortVarValue string
+		prometheusHostVarName  string
+		prometheusHostVarValue string
+		ops                    ExporterOptions
+		expectedAddr           string
+	}{{
+		name:                   "port from env var with no host set",
+		prometheusPortVarName:  prometheusPortEnvName,
+		prometheusPortVarValue: "9092",
+		ops:                    exporterOptions,
+		expectedAddr:           "0.0.0.0:9092",
+	}, {
+		name:                   "no port set with host from env var",
+		prometheusHostVarName:  prometheusHostEnvName,
+		prometheusHostVarValue: "127.0.0.1",
+		ops:                    exporterOptions,
+		expectedAddr:           "127.0.0.1:9090",
+	}, {
+		name:                   "port set and host set to empty string",
+		prometheusPortVarName:  prometheusPortEnvName,
+		prometheusPortVarValue: "",
+		prometheusHostVarName:  prometheusHostEnvName,
+		prometheusHostVarValue: "",
+		ops:                    exporterOptions,
+		expectedAddr:           "0.0.0.0:9090",
+	}, {
+		name:         "no port or host from the env",
+		ops:          exporterOptions,
+		expectedAddr: "0.0.0.0:9090",
+	}}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.prometheusPortVarName != "" {
+				os.Setenv(tc.prometheusPortVarName, tc.prometheusPortVarValue)
+				defer os.Unsetenv(tc.prometheusPortVarName)
+			}
+			if tc.prometheusHostVarName != "" {
+				os.Setenv(tc.prometheusHostVarName, tc.prometheusHostVarValue)
+				defer os.Unsetenv(tc.prometheusHostVarName)
+			}
+			mc, err := createMetricsConfig(context.Background(), tc.ops)
+			if err != nil {
+				t.Error(err)
```

Comment:
Should this be `t.Fatal`? I think if there's error, it's meaningless to use the returned `mc`.